### PR TITLE
[GEOS-10739] Contact information user interface feedback for welcome message

### PR DESCRIPTION
**Pull request by davidblasby**
[GEOS-10739](https://osgeo-org.atlassian.net/browse/GEOS-10739)

This is some simple GUI changes for the  Workspace Edit page.

There are some layout changes, but the biggest difference is that the "welcome" text went from an international textbox to an international textarea.

image

Notable changes;
1. "welcome" text is an international textarea (from textbox)
2. setting and services are now on the same column (were left/right)
3. services moved to BEFORE setting (consistent with the rest of the interface)
4. minor sizing changes so horizontal lines are the same length

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [x] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

**Comment by jodygarnett** [comment]
Thanks David, I am waiting in this change before updating docs.

**Review by jodygarnett — APPROVED** [review]

**Comment by jodygarnett** [comment]
Added some docs; giving my reputation with spelling some review is recommended.

I note that the workspace override of "global settings" lacks a header. Not sure what (if anything) can be done about that.

File: doc/en/user/source/configuration/contact.rst [hunk 1]
```diff
@@ -3,18 +3,46 @@
...
+Organization
+------------
+
+Complete this form with the organization details.
+
+.. figure:: img/contact_organization.png
+   
+   Contact Information Organization
+
+The welcome message is displayed in the welcome page header as an introduction to the GeoServer web services. The organization name and online resource, if provided, are used in the welcome page header to for the :guilabel:`For more information visit` link.
```
Comment (bradh) [inline doc/en/user/source/configuration/contact.rst:19]:
```suggestion
The welcome message is displayed in the welcome page header as an introduction to the GeoServer web services. The organization name and online resource, if provided, are used in the welcome page header for the :guilabel:`For more information visit` link.
```

File: doc/en/user/source/configuration/virtual-services.rst [hunk 2]
```diff
@@ -125,3 +127,62 @@ It is possible to completely restrict access to the global OWS services by setti
 To disable global services log into the GeoServer web administration interface and navigate to "Global Settings". Uncheck the "Enable Global Services" check box.
```
Comment (bradh) [inline doc/en/user/source/configuration/virtual-services.rst:127]:
```suggestion
To disable global services, log into the GeoServer web administration interface and navigate to "Global Settings". Uncheck the "Enable Global Services" check box.
```

File: doc/en/user/source/configuration/virtual-services.rst [hunk 3]
```diff
@@ -125,3 +127,62 @@ It is possible to completely restrict access to the global OWS services by setti
...
 
    .. figure:: img/global-services.jpg
+
+.. _workspace_isolated:
+
+Isolated Workspaces
+^^^^^^^^^^^^^^^^^^^
+
+Isolated workspaces content is only visible and queryable in the context of a virtual service bound to the isolated workspace. This means that isolated workspaces content will not show up in global capabilities documents and global services cannot query isolated workspaces contents. Is worth mentioning that those restrictions don't apply to the REST API.
```
Comment (bradh) [inline doc/en/user/source/configuration/virtual-services.rst:136]:
```suggestion
Isolated workspaces content is only visible and queryable in the context of a virtual service bound to the isolated workspace. This means that isolated workspaces content will not show up in global capabilities documents and global services cannot query isolated workspaces contents. Note that these restrictions do not apply to the REST API.
```

File: doc/en/user/source/configuration/virtual-services.rst [hunk 4]
```diff
@@ -125,3 +127,62 @@ It is possible to completely restrict access to the global OWS services by setti
 To disable global services log into the GeoServer web administration interface and navigate to "Global Settings". Uncheck the "Enable Global Services" check box.
 
    .. figure:: img/global-services.jpg
+
+.. _workspace_isolated:
+
+Isolated Workspaces
```
Comment (bradh) [inline doc/en/user/source/configuration/virtual-services.rst:133]: Not sure why this ended up in a PR about the welcome message.

File: doc/en/user/source/data/webadmin/workspaces.rst [hunk 5]
```diff
@@ -93,71 +87,86 @@ Once enabled clicking on the service link will open the settings page for the se
...
+'''''''''''''''
 
-An isolated workspace will be able to reuse a namespace already used by another workspace, but its resources (layers, styles, etc ...) can only be retrieved when using that workspace virtual services and will only show up in those virtual services capabilities documents.
+Clients accessing this workspace as a virtual service, or via the welcome page, will use the contact information provided here:
 
-It is only possible to create two or more workspaces with the same namespace in GeoServer if only one of them is non isolated, i.e. isolated workspaces have no restrictions in namespaces usage but two non isolated workspaces can't use the same namespace.
+* The :guilabel:`email` address if provided, will be used as the administrator contact in the welcome page footer for this workspace.
 
-The following situation will be valid:
+.. figure:: img/workspace_contaxt.png
```
Comment (bradh) [inline doc/en/user/source/data/webadmin/workspaces.rst:123]: The file name should have been `workspace_context.png`. However as long as they match....

File: doc/en/user/source/installation/upgrade.rst [hunk 6]
```diff
@@ -30,6 +30,29 @@ The general GeoServer upgrade process is as follows:
 Notes on upgrading specific versions
 ------------------------------------
 
+Log4J Upgrade (GeoServer 2.21 and newer)
```
Comment (bradh) [inline doc/en/user/source/installation/upgrade.rst:33]: Not sure this is relevant here either.

File: doc/en/user/source/data/webadmin/workspaces.rst [hunk 7]
```diff
@@ -46,40 +46,34 @@ To view or edit a workspace, click the workspace name. A workspace configuration
...
-^^^^^^^^^^^^^^^^^^
-
-Use :guilabel:`Enabled` checkbox to override the global contact information for this workspace.
-
-.. figure:: img/workspace_settings.png
-   
-   Enable workspace settings to provide default contact information
-   
-Clients accessing this workspace as a :ref:`virtual_services` will use the contact information provided here.
+  The worksapce contents will only visible and queryable in the context of a :ref:`virtual_services` as described below :ref:`workspace_isolated`.
```
Comment (bradh) [inline doc/en/user/source/data/webadmin/workspaces.rst:69]:
```suggestion
  The workspace contents will only be visible and queryable in the context of a :ref:`virtual_services` as described below :ref:`workspace_isolated`.
```

**Review by bradh — COMMENTED** [review]
@jodygarnett A few doc fixes, but it looks like you have some other stuff mixed in that perhaps wasn't intended.

**Review by aaime — COMMENTED** [review]
Found one minor issue. Code wise the changes are small, indeed most seem documentation, with a few off-topic changes.

One thing I've noticed, not a big deal, is that there is very little space between the services and the settings title in the workspace page:

image

**Comment by jodygarnett** [comment]
> @jodygarnett A few doc fixes, but it looks like you have some other stuff mixed in that perhaps wasn't intended.

I did add a note to the update section from the recent blog post. I hope it is not too much off topic as a documentation update.

**Comment by jodygarnett** [comment]
> Found one minor issue. Code wise the changes are small, indeed most seem documentation, with a few off-topic changes.
> 
> One thing I've noticed, not a big deal, is that there is very little space between the services and the settings title in the workspace page:

That *legend* element spacing seems to be a consistent trouble across the UI. It could be the amount of nesting and ul / li / div elements are unusual here.

I experiment with introducing an H3 Contact Information and H3 Other Settings and the result looked more consistent; be behaved in a difficult manner - stretching the settings legend width to "contain" the subsequent heading.

File: doc/en/user/source/configuration/virtual-services.rst (hunk 4 above)
Comment (jodygarnett) [reply doc/en/user/source/configuration/virtual-services.rst:133]: I did not write this, it was removed from the workspace reference page.

File: doc/en/user/source/installation/upgrade.rst (hunk 6 above)
Comment (jodygarnett) [reply doc/en/user/source/installation/upgrade.rst:33]: It is indeed off topic; I was going through the pages to see if workspace was mentioned. I will take this to a separate PR.

File: doc/en/user/source/data/webadmin/workspaces.rst [hunk 8]
```diff
@@ -46,40 +46,34 @@ To view or edit a workspace, click the workspace name. A workspace configuration
    
    Workspace named "topp"
    
-A workspace is defined by a name and a Namespace URI (Uniform Resource Identifier). The workspace name is limited to ten characters and may not contain space. A URI is similar to a URL, except URIs do not need to point to a actual location on the web, and only need to be a unique identifier. For a Workspace URI, we recommend using a URL associated with your project, with perhaps a different trailing identifier. For example, ``http://www.openplans.org/topp`` is the URI for the "topp" workspace. 
+A workspace is defined by a name and a Namespace URI (Uniform Resource Identifier).
 
-Root Directory for REST PathMapper 
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+* :guilabel:`Name`: The workspace name is limited to ten characters and may not contain space.
```
[the remark this answers is not in the document]
Comment (aaime) [reply doc/en/user/source/data/webadmin/workspaces.rst:51]: I think there is a misunderstanding here... while you're already modifying the page, you should remove the 10 chars limit from the docs. The possibility to actually limit that is long gone, too many existing data directories out there with a possibly longer name. Even the "release" data directory has one that goes beyond the limit ("it.geosolutions", even if it's not actually used by any store...)

**Comment by davidblasby** [comment]
> I experiment with introducing an H3 Contact Information and H3 Other Settings and the result looked more consistent; be behaved in a difficult manner - stretching the settings legend width to "contain" the subsequent heading.

I did a quick test of this and it looks like its working as expected.

File: doc/en/user/source/data/webadmin/workspaces.rst (hunk 8 above)
Comment (jodygarnett) [inline doc/en/user/source/data/webadmin/workspaces.rst:51]:
```suggestion
* :guilabel:`Name`: The workspace name, may not contain whitespace.
```

[the remark this answers is not in the document]
Comment (jodygarnett) [reply doc/en/user/source/data/webadmin/workspaces.rst]: You are correct I misunderstood. Doc updated to remove 10 char limit.

**Pull request merged**